xarray/coding/variables.py

```diff
@@ -216,7 +219,10 @@ def decode(self, variable, name=None):
 
 
 def _scale_offset_decoding(data, scale_factor, add_offset, dtype):
-    data = np.array(data, dtype=dtype, copy=True)
+    if not is_duck_array(data):
+        data = np.array(data, dtype=dtype, copy=True)
+    else:
+        data = data.astype(dtype=dtype, copy=True)
```

Comment:
Could we just use `data = data.astype(dtype=dtype, copy=True)` in every case, now that we cast indexing results to NumPy arrays?

Reply:
seems to work!